libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/tasks/Tasks.kt

```diff
@@ -560,13 +560,18 @@ abstract class KotlinCompile @Inject constructor(
             }
             project.extensions.extraProperties[TRANSFORMS_REGISTERED] = true
 
+            val buildMetricsReporterService = BuildMetricsReporterService.registerIfAbsent(project)
```

Comment:
Service registration does additional work, can you please add a method that fetches already registered service?

Reply:
I've updated `BuildMetricsReporterService.registerIfAbsent` (in a standalone commit to make it clearer) so that it returns the registered service immediately, it should be okay right?

Reply:
Yup. looks good.